surefire-providers/surefire-junit-platform/src/main/java/org/apache/maven/surefire/junitplatform/RunListenerAdapter.java

```diff
@@ -47,6 +47,7 @@
     implements TestExecutionListener
 {
     private final ConcurrentMap<TestIdentifier, Long> testStartTime = new ConcurrentHashMap<>();
+    private final ConcurrentMap<TestIdentifier, TestExecutionResult> failures = new ConcurrentHashMap<>();
```

Comment:
`failures` should be cleaned up when the tests finished.

Reply:
Clearing in `testPlanExecutionFinished` doesn't allow the data to be accessible when referenced in `JUnitPlatformProvider#invokeAllTests`. Having it clear per-test in `testPlanExecutionStarted` keeps the expected behavior.

Moving the `failures.clear()` as detailed resolves the ITS failures mentioned today:
![image](https://user-images.githubusercontent.com/21371686/65977509-0b4ac480-e440-11e9-8215-ff3a6b5543c9.png)

Reply:
but the tests may run in parallel. Clearing the Map concurrently would an issue for the methods in this adapter. WDYT?

Reply:
Perhaps associating a test context to the failures would be a sensible solution here. 

If the `failures` cannot be cleared in `testPlanExecutionFinished` due to them being needed in `JUnitPlatformProvider#invokeAllTests`, how should they be cleared? If done externally by the providers even the providers that don't need the `failures` collection would need to clear them. Any ideas?